File: fizzbuzz.cpp

#include <iostream>

using namespace std;

string fizzbuzz (int n) {
    bool d3 = n % 3 == 0;
    bool d5 = n % 5 == 0;
    if (d3 && d5) {
        return "FizzBuzz";
    } else if (d3) {
        return "Fizz";
    } else if (d5) {
        return "Buzz";
    } else {
        return to_string (n);
    }
}

int main () {
    for (int i = 1; i <= 100; ++i) {
        cout << fizzbuzz (i);
        if (i != 100) {
            cout << ", ";
        } else {
            cout << "." << endl;
        }
    }
}

File: fizzbuzz.hs

import Data.List (intercalate)

fizzbuzz :: Integer -> String
fizzbuzz n | d3 && d5  = "FizzBuzz"
           | d3        = "Fizz"
           | d5        = "Buzz"
           | otherwise = show n
           where d3 = n `mod` 3 == 0
                 d5 = n `mod` 5 == 0

main :: IO ()
main = putStrLn . (++".") . intercalate ", " $ map fizzbuzz[1..100]
